Connected System Template (CST) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/helloworld/templates/GoogleDriveCreateFolderIntegrationTemplate.java

```diff
@@ -0,0 +1,86 @@
+
+import static com.mycorp.helloworld.templates.GoogleDriveSendFileIntegrationTemplate.FILE_NAME_KEY;
+
+import java.io.IOException;
+
+import com.appiancorp.connectedsystems.simplified.sdk.SimpleIntegrationTemplate;
+import com.appiancorp.connectedsystems.simplified.sdk.configuration.SimpleConfiguration;
+import com.appiancorp.connectedsystems.templateframework.sdk.ExecutionContext;
+import com.appiancorp.connectedsystems.templateframework.sdk.IntegrationResponse;
+import com.appiancorp.connectedsystems.templateframework.sdk.TemplateId;
+import com.appiancorp.connectedsystems.templateframework.sdk.configuration.PropertyPath;
+import com.appiancorp.connectedsystems.templateframework.sdk.diagnostics.IntegrationDesignerDiagnostic;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+
+@TemplateId(name="GoogleDriveCreateFolderIntegrationTemplate")
+public class GoogleDriveCreateFolderIntegrationTemplate extends SimpleIntegrationTemplate {
+
+  private static final String FOLDER_MIME_TYPE = "application/vnd.google-apps.folder";
+
+  @Override
+  protected SimpleConfiguration getConfiguration(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      PropertyPath propertyPath,
+      ExecutionContext executionContext) {
+    return integrationConfiguration.setProperties(
+        //The same key (FILE_NAME_KEY) is used in the other template to create a textProperty for document name,
+        // therefore when you toggle between operations, the value you enter in this field will persist.
+        textProperty(FILE_NAME_KEY)
+            .label("Folder Name")
+            .isRequired(true)
+            .build()
+    );
+  }
+
+  @Override
+  protected IntegrationResponse execute(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      ExecutionContext executionContext) {
+    String folderName = integrationConfiguration.getValue(FILE_NAME_KEY);
+    File fileMetadata = new File().setName(folderName).setMimeType(FOLDER_MIME_TYPE);
+
+    CreateFolderDiagnosticsFactory createFolderDiagnosticsFactory = new CreateFolderDiagnosticsFactory(
+        connectedSystemConfiguration, integrationConfiguration);
+
+    GoogleCredential credential = new GoogleCredential().setAccessToken(executionContext.getAccessToken().get());
+    Drive drive = new Drive.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(), credential).build();
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder =  IntegrationDesignerDiagnostic.builder();
+
+    final long start = System.currentTimeMillis();
```

Comment:
I would just use a stopwatch here I think it is cleaner and it shows them how to do some dependencies